djstripe/models.py

```diff
@@ -2970,7 +2970,8 @@ def __str__(self):
 
     def update(
         self, plan=None, application_fee_percent=None, coupon=None, prorate=djstripe_settings.PRORATION_POLICY,
-        proration_date=None, metadata=None, quantity=None, tax_percent=None, trial_end=None
+        proration_date=None, metadata=None, quantity=None, tax_percent=None, trial_end=None,
+        cancel_at_period_end=False
```

Comment:
``djstripe_settings.CANCELLATION_AT_PERIOD_END``?

Reply:
Valid point. Question is, why would one want to update a subscription that is set to cancel at period end? Can you think of a valid reason? I do see a reason for having a default for `djstripe_settings.PRORATION_POLICY`, because that is very important to have, when users want to change their subscription plan. But I'm not sure, we should be using `djstripe_settings.CANCELLATION_AT_PERIOD_END` in this context - actually I feel very strongly against it, and here's why - if we have this setting set to True by default (which makes a good default for this setting for a subscription based service), then if users forgets to explicitly set this parameter to False, any updates to user's subscription, will set their cancel_at_period_end to True - effectively cancelling their subscription. I think this setting only makes sense in the context of subscription cancellation: `def cancel(self, at_period_end=djstripe_settings.CANCELLATION_AT_PERIOD_END)`

Reply:
@tkrevh I'll be honest, I've just left this here as a "take a look at this tomorrow" note :). I haven't yet taken the time to go through the codebase and do the research. Your reasoning is sound; I'm wondering if there was a reason we explicitly left out the ``cancel_at_period_end`` argument

Reply:
The only reason I could think of, would be to prevent resurrection of cancelled subscriptions. But this could cause issues if user would want to reactivate their subscription (in our experience not unheard of) - i.e. we'd have to manually cancel the subscription with immediate effect and create a new one. If Stripe supports subscription resurrection, I think dj-stripe could as well. We're using a fork of dj-stripe with the proposed change exactly for this reason. We were using dj-stripe v1.0.1 before, and after upgrading to Python3/Django2, we upgraded to v1.2 and there we stumbled upon this problem. In v1.0.1 this was working.

Reply:
Ah well if it was working in 1.0.1 and now not working, we can assume this is possibly a regresssion

fyi @therefromhere